### PR TITLE
Make the site generator more tolerant of bad/unexpected data

### PR DESCRIPTION
**Pull request by leonardr**

### CODE REVIEW

File: bin/excel.py [hunk 1]
```diff
@@ -16,7 +16,10 @@ def dictreader(workbook, sheet_name):
             if first_row:
                 keys.append(cell.value)
             else:
-                d[keys[i]] = cell.value
+                if i >= len(keys):
+                    continue
```
Comment (leonardr) [inline bin/excel.py]: Without this, any content in a cell that has no label will crash the script.

File: bin/generate_landing_page [hunk 2]
```diff
@@ -141,6 +141,10 @@ class GenerateLandingPageScript:
 
           metadata = {}
 
+          if not lanes:
+              raise ValueError(
+                  "Could not find any lanes in the input. Something is clearly wrong; will not continue."
+              )
```
Comment (leonardr) [inline bin/generate_landing_page]: Some error conditions in parsing the spreadsheet caused it to look like there were no lanes, and the errors were unhelpful. If that should ever happen again, it's best to give an error message that points to the real problem.

File: bin/generate_landing_page [hunk 3]
```diff
@@ -264,6 +268,7 @@ class GenerateLandingPageScript:
           category_nav_404 += "<li><a href='/about.html'>About</a></li>"
 
           # Generate the book lanes and insert a call to action for SimplyE on the homepage
+          cacheManifest = "CACHE MANIFEST\n #timestamp "
```
Comment (leonardr) [inline bin/generate_landing_page]: This variable was being initialized inside a loop that wasn't guaranteed to run.

File: bin/model.py [hunk 4]
```diff
@@ -488,7 +488,7 @@ def from_workbook(cls, workbook, allowable_sources, penguin_client):
                 featurable = True
             book.featurable = featurable
 
-            mta_category = row.get(Book.MTA_CATEGORY_1)
+            mta_category = row.get(Book.MTA_CATEGORY_1).strip()
```
Comment (leonardr) [inline bin/model.py]: Without this code, we had both a "New and Notable" lane and a "New and Notable " lane.

File: bin/generate_landing_page [hunk 5]
```diff
@@ -306,7 +311,6 @@ class GenerateLandingPageScript:
                   "books": "".join(book.html_category for book in sorted_books)
               }
 
-              cacheManifest = "CACHE MANIFEST\n #timestamp "
               cacheManifest += str(datetime.datetime.utcnow())
               cacheManifest += "\n\n"
```
Comment (aslagle) [inline bin/generate_landing_page]: you should move up these next two lines as well, you wouldn't want a cache manifest without a timestamp.

**Comment by aslagle** [comment]
you have an extraneous submodule update in here. It means this branch won't work without the changes I have in #221.

**Pull request merged**